tests/filecheck/dialects/func/func_ops.mlir

```diff
@@ -31,10 +39,37 @@
    // CHECK-NEXT:   "func.return"(%1) : (!test.type<"int">) -> ()
    // CHECK-NEXT: }) {"sym_name" = "arg_rec", "function_type" = (!test.type<"int">) -> !test.type<"int">, "sym_visibility" = "private"} : () -> ()
 
+  func.func @cus_arg_rec(%0 : !test.type<"int">) -> !test.type<"int"> {
+    %1 = "func.call"(%0) {"callee" = @cus_arg_rec} : (!test.type<"int">) -> !test.type<"int">
+    "func.return"(%1) : (!test.type<"int">) -> ()
+  }
+
+   // CHECK:      "func.func"() ({
+   // CHECK-NEXT: ^{{.*}}(%{{.*}} : !test.type<"int">):
+   // CHECK-NEXT:   %{{.*}} = "func.call"(%{{.*}}) {"callee" = @cus_arg_rec} : (!test.type<"int">) -> !test.type<"int">
+   // CHECK-NEXT:   "func.return"(%{{.*}}) : (!test.type<"int">) -> ()
+   // CHECK-NEXT: }) {"sym_name" = "cus_arg_rec", "function_type" = (!test.type<"int">) -> !test.type<"int">, "sym_visibility" = "private"} : () -> ()
+
+  func.func @cus2_arg_rec(!test.type<"int">) -> !test.type<"int"> {
```

Comment:
I was... not aware of that syntax.
Do you know why MLIR accepts it? It looks quite weird no?

Reply:
What do you mean? The anonymous argument?

Reply:
Yeah, I had no clue why this was accepted. It seems weird no?

Reply:
Oh, with the block arguments.
Yeah, it is really weird, but hey, why not?
I would argue it can be of use if you do dark text-based IR transformations? :thinking: I don't know about any real use case! But it is accepted :sweat_smile: